pages/views.py

```diff
@@ -45,6 +45,28 @@ def contact_view(request):
     return render(request, 'pages/contact.html', {})
 
 
+def send_details_view(request):
+    personal_details = {}
+
+    personal_details['name'] = request.POST.get('name')
+    personal_details['number'] = request.POST.get('number')
+    personal_details['email'] = request.POST.get('email')
+    message = request.POST.get('message')
+
+    email_details(personal_details, message)
+
+    return redirect(request.POST.get('origin'))
+
+
+def email_details(personal_details, message):
+
+    for key in personal_details.keys():
+        if personal_details[key] in [None, ' ']:

```

Comment:
`' '` is not an empty string, it has a space.... Empty string is `''` which is the value there in the field if it is not filled.

In addition, consider using a _tuple_ instead of a _list_ here:

```python
if personal_details[key] in (None, '')
```

Please check your code....

Reply:
Yeah sure, but I didn't check it cuz this function is not fucking implemented...